### DISCUSSION TITLE
New documentation template

### DISCUSSION THREAD

**Discussion by mfsilva22** [PrairieLearn/PrairieLearn#3619] [Dev]
Trying to get some ideas following the suggestions in Issue #3612. After reading https://documentation.divio.com, and the documentations that use this structure, I think we can make some small changes to the names, and have the following structure for our documentation:

- Get started
- How-to guides
- Reference guides
- Background

See below a more detailed description, using the current documentation files. We will need to move a lot of content around, to fit better the new structure.

I am starting this discussion here, to make sure we don't lose old messages on Slack. Let me know if you have suggestions/comments. I will be likely posting comments here, as I start working on each section, and questions start to arise.

--------------

# Instructors Manual
## Get started
(This will have a step-by-step process to get a new user started with PrairieLearn. It will only describe a simple question and a simple assessment. The question gallery will include one example of each question type, with detailed explanations for each question. In the workshop content, new users can create more variations of questions and assessments at their own pace. The video will be included for reference. We can later record better videos.)

### Assessment types
(This will include guides on how to setup many different assessment types (including remote exams, sync/async, timed, etc)

## Reference guides
(Reference guides have one job only: to describe. They are information-oriented)

## Background
(this section will have information about PL background, architecture, and case studies that used PL. We will use this to include reference to the papers)

### The use of auto-graded drawing tools in engineering courses

# Students Manual

## Reference guides

# Developers Manual

# Admin Manual

**Comment by mwest1066** [comment]
This looks great! 🎉  I don't have any other changes to suggest at this point. I'm really looking forward to seeing the newly reorganized docs and the new material in "Get started" and "How-tos"!

**Comment by mfsilva22** [comment]
I am thinking of creating the following Directory structure for the new documentation:

docs
+-- Instructors
|   +-- GetStarted
|   +-- HowTos
|   +-- Reference
|   +--Background
+-- Students
... same structure as others

The images now are all saved in the main docs folder. I am thinking of creating separate folders for the images as well. Maybe something like this?

docs
+-- Instructors
|   +-- images
|   +-- GetStarted
|   `-- requestCourse.md`
|   +-- HowTos
|   +-- Reference
|   +--Background

Or do you prefer if we have just one `images` folder?

I am not too familiar with markdown and websites, but suppose that we use the structure above, how do I add an image in the `requestCourse.md` file?

![](../images/requestCourseButton.png) --> does not seem to work

**Reply by mwest1066** [reply]
> The images now are all saved in the main docs folder. I am thinking of creating separate folders for the images as well. Maybe something like this?

I generally find it best to keep the images next to the pages they belong to. One strategy to do this is to put each page in its own directory (as `index.md`) with the images next to it. That is, something like:

```
docs
+-- Instructors
|   +-- GetStarted
|   |   +-- RequestCourse
|   |   |   +-- index.md
|   |   |   +-- courseRequestImage.png
|   |   |
|   |   +-- CreateFirstQuestion
|   |       +-- index.md
|   |       +-- questionButton.png
|   |
|   +-- HowTos
|   ...
|
+-- Developers
...
```

The main disadvantage of this arrangement is that you end up with lots of files, but the advantage is that it stays organized and it's always very clear which files belong to which pages.

This would also fix your problem with referencing images, because everything would always be in the same directory.

**Reply by mfsilva22** [reply]
can we go with an intermediate level of details, so that some folders can contain more than one markdown file (for example, the `QuestionGallery` folder
```
docs
+-- Instructors
|   +-- GetStarted
|   |   +-- RequestCourse
|   |   |   +-- index.md
|   |   |   +-- courseRequestImage.png
|   |   +-- CreateCourseContent
|   |       +-- index.md
|   |       +-- questionButton.png
|   |   +-- QuestionGallery
|   |       +-- index.md
|   |       +-- checkbox.md
|   |       +-- inputnumber.md
|   |       +-- image1.png
|   +-- HowTos
|   ...
|
+-- Developers
...
```

**Reply by mwest1066** [reply]
For sure, I'm not dogmatic about the one-directory-per-page structure. It's fine to have multiple related pages all collapsed into one directory if that makes sense.

I'm slightly skeptical that it's going to be tidier all in one directory, however 😄
